### PR TITLE
Fix most of the compile errors in the client package

### PR DESCRIPTION
**Pull request by tterrag1098**
The remaining errors are caused by missing patches to Block (waiting on @illyohs) and missing patches to KeyBinding (which I think @mezz is handling).

### CODE REVIEW

**Review by mezz — APPROVED** [review]

File: src/main/java/net/minecraftforge/fml/client/config/GuiConfigEntries.java [hunk 1]
```diff
@@ -1684,7 +1684,7 @@ private String removeTag(String target, String tagStart, String tagEnd)
     /**
      * Provides an interface for defining GuiConfigEntry.listEntry objects.
      */
-    public static interface IConfigEntry extends GuiListExtended.IGuiListEntry
+    public static interface IConfigEntry
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/fml/client/config/GuiConfigEntries.java:1687]: Whats the replacement?

File: src/main/java/net/minecraftforge/common/extensions/IForgeDimension.java [hunk 2]
```diff
@@ -18,4 +23,19 @@
     {
         return null;
     }
+
+    @Nullable
+    IRenderHandler getSkyRenderer();
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/common/extensions/IForgeDimension.java]: @Dist?

File: src/main/java/net/minecraftforge/client/resource/SelectiveReloadStateHandler.java [hunk 3]
```diff
@@ -58,7 +59,7 @@ public void beginReload(Predicate<IResourceType> resourcePredicate)
      */
     public Predicate<IResourceType> get()
     {
-        if (this.currentPredicate == null || !ForgeModContainer.selectiveResourceReloadEnabled)
+        if (this.currentPredicate == null) // TODO configs || !ForgeMod.selectiveResourceReloadEnabled)
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/client/resource/SelectiveReloadStateHandler.java]: Default to off until config is back

File: src/main/java/net/minecraftforge/client/IClientCommand.java [hunk 4]
```diff
@@ -33,7 +33,8 @@
      * @param sender the command sender
      * @param message the message, without potential prefix
      * @return true to allow the usage of this command without the prefix
-     */
+     * /
     boolean allowUsageWithoutPrefix(ICommandSender sender, String message);
 
 }
+*/
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/client/IClientCommand.java]: Have you looked into the new command system? Thoughts on replacements? TODO atleast...

File: src/main/java/net/minecraftforge/fml/client/config/GuiConfigEntries.java (hunk 1 above)
Comment (tterrag1098) [reply src/main/java/net/minecraftforge/fml/client/config/GuiConfigEntries.java:1687]: There isn't one, that interface became an abstract class so we have to extend it lower in the hierarchy.

File: src/main/java/net/minecraftforge/common/extensions/IForgeDimension.java (hunk 2 above)
Comment (tterrag1098) [reply src/main/java/net/minecraftforge/common/extensions/IForgeDimension.java]: IRenderHandler itself (which btw is not an interface O.o) isn't client only, so it's not strictly needed here.

**Pull request merged**